magento/magento2
# magento/magento2#36431: Undefined index in TierPriceValidator.php

- fixed issue with undefined index customer group for changing tier price





### Description (*)

The problem was due to the fact that the TierPriceValidator::retrieveGroupValue() returned an unknown index for the customer group instead of a false value (if the customer group is undefined), so when changing the price through the API, we received an index error instead of an exception

### Related Pull Requests


### Fixed Issues (if relevant)

1. Fixes magento/magento2#<issue_number>

### Manual testing scenarios (*)

1. Create a customer group with a trailing space in it's name.
2. Check the customer group in the admin panel (it wouldn't contain space before or after the customer group name) 
3. Create admin integration token using POST: /rest/default/V1/integration/admin/token
4. Using the API endpoint /rest/V1/products/tier-prices, attempt to POST a tier price for the new customer group but with the trailing space.
Payload:
{
"prices": [
{
"price": 40,
"price_type": "fixed",
"website_id": 0,
"sku": "24-MB01",
"customer_group": " your customer group name ",
"quantity": 1,
"extension_attributes": {}
}
]
}
This should return exceptions like ''No such entity with Customer Group = %customerGroup ... " instead of  error message like Message: Notice: Undefined index: XXXXX in ~/public_html/vendor/magento/module-catalog/Model/Product/Price/Validation/TierPriceValidator.php on line 474"

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves [magento/magento2#37523: magento/magento2#36431: Undefined index in TierPriceValidator.php](https://github.com/magento/magento2/issues/36431)
